.github/workflows/fork-connector-tests.yml

```diff
@@ -0,0 +1,138 @@
+              "gitref": "${{ github.event.pull_request.head.ref }}",
+              "pr": "${{ github.event.pull_request.number }}"
+            }
+
+      - name: Post success commit status
+        if: steps.trigger-tests.outcome == 'success' && steps.trigger-tests.outputs.workflow-conclusion == 'success'
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "success",
+              "context": "Fork Connector Tests",
+              "description": "Connector tests passed in fork with BYO secrets",
+              "target_url": "${{ steps.trigger-tests.outputs.workflow-url }}"
+            }'
+
+      - name: Post failure commit status
+        if: steps.get-app-token.outcome == 'success' && (steps.trigger-tests.outcome == 'failure' || steps.trigger-tests.outputs.workflow-conclusion != 'success')
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "failure",
+              "context": "Fork Connector Tests",
+              "description": "Connector tests failed in fork with BYO secrets",
+              "target_url": "${{ steps.trigger-tests.outputs.workflow-url || format(''{0}/{1}/actions/runs/{2}'', github.server_url, github.repository, github.run_id) }}"
+            }'
+
+      - name: Comment on PR with results
+        if: steps.get-app-token.outcome == 'success'
+        uses: peter-evans/create-or-update-comment@v4
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            ## Fork Connector Tests Results
+
+            ${{ steps.trigger-tests.outputs.workflow-conclusion == 'success' && '✅ Tests passed' || '❌ Tests failed' }}
+
+            The connector tests ran in your fork's repository context using your BYO secrets.
+
+            [View test results](${{ steps.trigger-tests.outputs.workflow-url }})
+
+      - name: Skip message for non-external forks
+        if: steps.check-fork.outputs.is-external-fork != 'true'
+        run: |
+          echo "Skipping fork connector tests - PR is not from an external fork"
+
+      - name: Skip message for missing GitHub App
+        if: steps.check-fork.outputs.is-external-fork == 'true' && steps.get-app-token.outcome != 'success'
+        run: |
+          echo "Skipping fork connector tests - GitHub App not installed on fork"
+          echo "Fork owner '${{ steps.fork-owner.outputs.owner }}' needs to install the Airbyte GitHub App"
```

Comment:
Put any "messaging" (aka printing to the job log) immediately after the step to get github app creds.